app/models/physical_storage.rb

```diff
@@ -46,6 +47,41 @@ def refresh_ems
     EmsRefresh.queue_refresh(ext_management_system)
   end
 
+  def raw_delete_physical_storage
+    raise NotImplementedError, _("raw_delete_physical_storage must be implemented in a subclass")
+  end
+
+  def delete_physical_storage
+    raw_delete_physical_storage
+  end
+
+  supports :delete do
+    unless respond_to?(:delete)
```

Comment:
Side note, `#delete` is an ActiveRecord method so every model instance will respond to `:delete`